### ISSUE TITLE
Different postprocessings

### ISSUE THREAD

**Issue by NPetsky** [google/automl#832]
There are currently a lot of different postprocessings as far as I can see and I wanted to ask what are the differences between them and which is the best one (soft, hard, combined...) ?
Also is there still a difference between the tensorflow nms and the numpy one? And if yes which one is preferable?
I also discovered there are a few functions in nms_np.py which doesn't seem to be used any more ( _generate_detections, decode_box_outputs_np, sigmoid) so maybe they can be removed?
Thanks for your help!

**Comment by mingxingtan** [comment]
Hi @NPetsky 

The numpy version of soft-nms has the best coco validation accuracy, but it is also the slowest one.

Unfortunately, there is still a small difference  between TF and numpy NMS  (numpy per_class_nms is about ~0.2AP better than TF version). Despite that, I recommend using TF version global nms for inference, as it is much faster.

**Comment by NPetsky** [comment]
Thank you!
